src/pure.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { ReactElement, ReactNode, Suspense } from 'react'
+import { act, create, ReactTestRenderer } from 'react-test-renderer'
+import asyncUtils from './asyncUtils'
+import { cleanup, addCleanup, removeCleanup } from './cleanup'
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type Props<T = any, R = any> = {
```

Comment:
In general, I prefer generic types to better describe what they are genericizing, e.g. `TProps` and `TResult` make following their usage (and intellisense) much easier to me.